src/gui/widgets/tab_container.hpp

```diff
@@ -0,0 +1,144 @@
+/*
+	Copyright (C) 2024
+	by Subhraman Sarkar (babaissarkar) <suvrax@gmail.com>
+	Part of the Battle for Wesnoth Project https://www.wesnoth.org/
+
+	This program is free software; you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation; either version 2 of the License, or
+	(at your option) any later version.
+	This program is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY.
+
+	See the COPYING file for more details.
+*/
+
+#pragma once
+
+#include "gui/widgets/container_base.hpp"
+#include "gui/widgets/listbox.hpp"
+#include "gui/core/window_builder.hpp"
+
+namespace gui2
+{
+
+namespace implementation
+{
+struct builder_tab_container;
+}
+
+/**
+ * Class for a tabbed container
```

Comment:
This first line can be deleted, it's already said be the machine-readable part.

Reply:
Removed.